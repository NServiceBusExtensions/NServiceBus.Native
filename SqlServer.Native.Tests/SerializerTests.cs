using System.Collections.Generic;
using ObjectApproval;
using Xunit;

public class SerializerTests
{
    [Fact]
    public void Dictionary()
    {
        var serialized = Serializer.SerializeDictionary(
            new Dictionary<string, string>
            {
                {"key", "value"},
                {@"a\b", @"a\b"},
                {@"a\\b", @"a\\b"},
                {"a\"b", "a\"b"},
                {"a/b", "a/b"},
                {"a//b", "a//b"},
                {@"a\/b", @"a\/b"},
            });
        ObjectApprover.VerifyWithJson(Serializer.DeSerializeDictionary(serialized));
    }

    [Fact]
    public void List()
    {
        var serialized = Serializer.SerializeList(
            new List<string>
            {
                "value",
                @"a\b",
                @"a\\b",
                "a\"b",
                "a/b",
                "a//b",
                @"a\/b",
            });
        ObjectApprover.VerifyWithJson(Serializer.DeSerializeList(serialized));
    }
}