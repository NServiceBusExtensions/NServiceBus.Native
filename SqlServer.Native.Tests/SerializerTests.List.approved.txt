{
  "$type": "List<string>",
  "$values": [
    "value",
    "a\\b",
    "a\\\\b",
    "a\"b",
    "a/b",
    "a//b",
    "a\\/b"
  ]
}